### PR TITLE
s2k: output valid private key data when in S2K GNU Dummy mode

### PR DESCRIPTION
**Pull request by maxtaco**
- expand test coverage to: import, export, import S2K GNU dummy data
- no encrypted flag for s2k dummy
- add our attached sign function into Go's PGP library
- run go fmt
- add a new key to play around with
- make detached sign use the right signing key
- new tests

### CODE REVIEW

File: openpgp/packet/private_key.go [hunk 1]
```diff
@@ -129,20 +128,32 @@ func (pk *PrivateKey) Serialize(w io.Writer) (err error) {
...
-		return
+	if pk.PrivateKey == nil {
+		_, err = buf.Write([]byte{
+			254,           // SHA-1 Convention
+			9,             // Encryption scheme (AES256)
+			101,           // GNU Extensions
+			2,             // Hash value (SHA1)
+			'G', 'N', 'U', // "GNU" as a string
+			1, // Extension type 1001 (minus 1000)
+		})
```
Comment (patrickxb) [inline openpgp/packet/private_key.go:142]: is there a document somewhere that describes this?  I can't find it here https://tools.ietf.org/html/rfc4880

Comment (maxtaco) [inline openpgp/packet/private_key.go:142]: it's 100% undocumented!

**Comment by patrickxb** [comment]
LGTM

Would be nice to have link to reference for S2K header.

File: openpgp/packet/private_key.go (hunk 1 above)
Comment (patrickxb) [inline openpgp/packet/private_key.go:142]: wonderful...ok.

**Pull request merged**